crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -596,19 +596,16 @@ impl kitsune2_api::KitsuneHandler for HolochainP2pActor {
                 )?);
             }
 
-            // if they sent us agents, spawn a task to insert them into the store
+            // if they sent us agents, insert them into the store
             if !agents.is_empty() {
                 let kitsune = self.kitsune.clone();
-                tokio::task::spawn(async move {
-                    for agent in agents {
-                        let space = match kitsune.space_if_exists(agent.space.clone()).await {
-                            None => continue,
-                            Some(space) => space,
-                        };
-                        space.peer_store().insert(vec![agent]).await?;
-                    }
-                    K2Result::Ok(())
-                });
+                for agent in agents {
```

Comment:
Kitsune's blocks module depends on agent infos being exchanged during preflight messaging, so this should no longer happen in a separate task, but the current task should block for this agent insertion to complete.